firedrake/supermeshimpl.pyx

```diff
@@ -98,3 +115,69 @@ def assemble_mixed_mass_matrix(V_A, V_B, candidates,
 
     CHKERR(MatAssemblyBegin(mat.mat, MAT_FINAL_ASSEMBLY))
     CHKERR(MatAssemblyEnd(mat.mat, MAT_FINAL_ASSEMBLY))
+
+
+def intersection_finder(mesh_A, mesh_B):
+    # Plan:
+    # Call libsupermesh_sort_intersection_finder_set_input
+    # Call libsupermesh_sort_intersection_finder_query_output
+    # Call libsupermesh_sort_intersection_finder_get_output
+    # Return the output
+
+    cdef:
+        numpy.ndarray[long, ndim=2, mode="c"] vertex_map_A, vertex_map_B
+        numpy.ndarray[double, ndim=2, mode="c"] vertices_A, vertices_B
+        long nindices
+        numpy.ndarray[long, ndim=1, mode="c"] indices, indptr
+        long nnodes_A, nnodes_B, ncells_A, ncells_B
+        int dim_A, dim_B, loc_A, loc_B
+
+    dim = mesh_A.geometric_dimension()
+    assert dim == mesh_B.geometric_dimension()
+    assert dim == mesh_A.topological_dimension()
+    assert dim == mesh_B.topological_dimension()
+
+    assert mesh_A.coordinates.function_space().ufl_element().degree() == 1
+    assert mesh_B.coordinates.function_space().ufl_element().degree() == 1
+
+    if mesh_A.comm.size > 1:
+        compatible = False
+        assert mesh_B._parallel_compatible is not None, "Whoever made mesh_B should explicitly mark mesh_A as having a compatible parallel layout."
+        for _mesh_A in mesh_B._parallel_compatible:
+            if mesh_A is _mesh_A():
+                compatible = True
+                break
+
+        if not compatible:
+            assert ValueError("Whoever made mesh_B should explicitly mark mesh_A as having a compatible parallel layout.")
+
+    vertices_A = mesh_A.coordinates.dat.data_ro_with_halos
+    vertices_B = mesh_B.coordinates.dat.data_ro_with_halos
+    vertex_map_A = mesh_A.coordinates.cell_node_map().values_with_halo.astype(long)
+    vertex_map_B = mesh_B.coordinates.cell_node_map().values_with_halo.astype(long)
```

Comment:
What is the actual Fortran type you need? Prefer explicit `numpy.int32/numpy.int64` if possible?

Reply:
The actual Fortran type is `c_long`. I think it's machine-dependent how many bits that has? I agree this type stuff is a mess, though.

Reply:
`numpy.long`?

Reply:
I guess so, yes.

Reply:
OK, done.

Reply:
Thanks.